peterpy/middlewares.py

```diff
@@ -0,0 +1,33 @@
+from aiohttp import web
+import logging
+
+from peterpy.database.connection import DatabaseConnection, DatabaseSession
+from peterpy.repositories.database_product_repository import DatabaseProductRepository
+from peterpy.services.product_service import ProductService
+
+
+def db_session_wrapper_factory(db_connection: DatabaseConnection):
+    @web.middleware
+    async def db_session_wrapper(request, handler):
+        logging.debug("db_session_wrapper called")
+        engine = db_connection.engine()
+
+        with DatabaseSession(engine) as session:
+            try:
+                repository = DatabaseProductRepository(session)
+                product_service = ProductService(repository)
+                request["product_service"] = product_service
+
+                response = await handler(request)
+                logging.debug("db_session_wrapper committing")
+                session.commit()
+                logging.debug("db_session_wrapper finished")
+                return response
+            except Exception as e:
+                logging.exception(
+                    "Exception happened in db_session_wrapper 1: %s - rolling back", e
+                )
+                session.rollback()
```

Comment:
rollback happens now, but is it correct I get Traceback error in my console, I'm catch every exception ? 
```
2024-05-17 08:39:21 - root - DEBUG - Adding product Product(product_id=UUID('a5513da1-afc8-4d41-9c88-7d0e8ca95ea8'), name='passing as part of list 9', price=2.576)
2024-05-17 08:39:21 - root - ERROR - Exception happened in db_session_wrapper 1: Product with name 'passing as part of list 4' already exists - rolling back
Traceback (most recent call last):
  File "/Users/peter.van.dijk/sideprojects/road-to-backend/peterpy/peterpy/middlewares.py", line 21, in db_session_wrapper
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter.van.dijk/sideprojects/road-to-backend/peterpy/peterpy/handlers/products.py", line 95, in add_products
    product = await product_service.add(name, price)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter.van.dijk/sideprojects/road-to-backend/peterpy/peterpy/services/product_service.py", line 15, in add
    raise ValueError(f"Product with name '{name}' already exists")
ValueError: Product with name 'passing as part of list 4' already exists
2024-05-17 08:39:21 - root - INFO - Closing database session
2024-05-17 08:39:21 - aiohttp.access - INFO - 127.0.0.1 [17/May/2024:07:39:21 +0100] "POST /products/ HTTP/1.1" 500 237 "-" "PostmanRuntime/7.39.0"
```